### PR TITLE
fix(ui5-tabcontainer): update tab strip on tab selection

### LINKED ISSUE #5172: TabContainer: Selecting tab programmatically, does not refresh selected tabs in the toolbar and the overflowButtons
**Describe the bug**
Selecting tab programmatically, does not refresh selected tabs in toolbar and overflowButtons.

**Isolated Example**
https://codesandbox.io/s/jovial-cloud-zilwwr?file=/src/App.js

**To Reproduce**
1. Go to https://codesandbox.io/s/jovial-cloud-zilwwr?file=/src/App.js
2. Click on Button "Tab 5"
3. Tab 5 content is now visible and selected but the toolbar and the overflowButtons didn´t refresh the tab selection

**Expected behavior**
If you select a tab programmatically, the toolbar and the overflowButtons should refresh too and should show the currently selected tab visually.

**Screenshots**

**UI5 Web Components for React Information**
`@ui5/webcomponents` version: Latest
`@ui5/webcomponents-react` version: Latest
Operating System: Windows
Browser: Chrome

**Additional context**

### PR DESCRIPTION
**Pull request by gmkv**
- Tabs are rearranged on the tab strip when a tab is selected programmatically.
- Updated documentation for getTabInStripDomRef to clarify use
with nested tabs.
- Refactored component and stabilized keyboard navigation

Fixes #5172
Part of #5116

### CODE REVIEW

File: packages/main/src/TabSeparator.js [hunk 1]
```diff
@@ -58,6 +58,14 @@ class TabSeparator extends UI5Element {
 		return true;
 	}
 
+	/**
+	 * Returns the DOM reference of the separator that is placed in the header.
```
Comment (BorisDafov) [inline packages/main/src/TabSeparator.js:62]: I suggest to combine into one note:
	 * <b>Note:</b> Tabs and separators, placed in the <code>subTabs</code> slot of other tabs are not shown in the header. Calling this method on such tabs or separators will return <code>undefined</code>.

File: packages/main/test/pages/TabContainer.html [hunk 2]
```diff
@@ -726,6 +726,44 @@ <h2>Text only Start And End Overflow Custom Overflow Buttons</h2>
...
+					<ui5-tab id="nestedTab" slot="subTabs" text="Nested tab">
+						<ui5-button>Button 1</ui5-button>
+					</ui5-tab>
+				</ui5-tab>
+				<ui5-tab id="lastTab" text="Eighth tab">
+					<ui5-button>Button 1</ui5-button>
+				</ui5-tab>
+			</ui5-tabcontainer>
+		
+			<ui5-button id="selectFirst">Select first tab</ui5-button>
```
Comment (BorisDafov) [inline packages/main/test/pages/TabContainer.html:763]: I suggest title case for button labels:
			<ui5-button id="selectFirst">Select First Tab</ui5-button>
			<ui5-button id="selectLast">Select Last Tab</ui5-button>
			<ui5-button id="selectNested">Select a Nested Tab</ui5-button>

**Review by BorisDafov — CHANGES_REQUESTED** [review]

**Review by BorisDafov — DISMISSED** [review]

**Review by dimovpetar — CHANGES_REQUESTED** [review]
Sub tabs menus cannot be opened the sample "Nested Tabs".
Try to press the arrow of tab "Two" or "Three"
image

**Review by dimovpetar — CHANGES_REQUESTED** [review]
When the overflow is not visible, expanding tabs with arrow down or click doesn't work

**Review by TeodorTaushanov — APPROVED** [review]

**Pull request merged**